### PR TITLE
fix(fabricx): restore endpoint parser for id=<group>,<type>,<host:port>

### PR DESCRIPTION
**Pull request by tinystone007**
## Problem

The fabric-x official ansible collection (`LF-Decentralized-Trust-labs/fabric-x-ansible-collection`) and the fabric-x-tool `config-builder` emit orderer endpoints in channel config as:

- `id=<group>,broadcast,<host>:<port>` for routers
- `id=<group>,deliver,<host>:<port>` for assemblers

See [ansible collection configtx template](https://github.com/LF-Decentralized-Trust-labs/fabric-x-ansible-collection/blob/main/roles/configtxgen/templates/configtx.yaml.j2).

FSC v0.8.2 parsed this format in `platform/fabricx/core/membership/endpoint.go` and filtered to broadcast-only endpoints in `Service.OrdererConfig` (https://github.com/hyperledger-labs/fabric-smart-client/blob/v0.8.2/platform/fabricx/core/membership/membership.go#L170-L207). The parser was removed during the v0.9/v0.10 membership refactor, which now treats every endpoint (including deliver-typed assembler ports) as broadcast.

## Impact

With the official configtx template, FSC clients attempt to `Broadcast` to assemblers (which only serve `Deliver`) and get:

```
failed to get status after broadcast to [host:7x53]: rpc error: code = Unknown desc = should not be used
```

Any FSC-based token-sdk / fabricx application deployed on top of the ansible-provisioned fabric-x infrastructure hits this regression on v0.9+.

## Fix

This PR ports the v0.8.2 `endpoint.go` and reintroduces the filter in `OrdererConfig`:

1. Parses `id=<group>,<type>,<host:port>` when present
2. Skips `deliver`-typed endpoints (clients only broadcast here; deliver is sourced from the local sidecar / committer)
3. Falls back to treating plain `<host:port>` as broadcast for backward compatibility with non-fabric-x callers

## Sibling PR

Related upstream regression patch: #1249 (arma -> BFT broadcaster alias, also from the same v0.9/v0.10 membership refactor).

## Testing

Verified end-to-end on a fabric-x v0.1.9 / orderer v0.0.24 deployment using the ansible collection configtx format:
- TMS deploy finality succeeds
- IssueToken / Transfer / Redeem full token lifecycle passes
- No more "should not be used" broadcast errors on assembler ports

### CODE REVIEW

File: platform/fabricx/core/membership/endpoint.go [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+	Type     string
+	Endpoint string
+}
+
+// parseEndpoint parses the fabric-x channel-config endpoint format
+// `id=<group>,(broadcast|deliver),<host>:<port>` as emitted by the official
+// fabric-x ansible collection and fabric-x-tool config-builder. A plain
+// `<host>:<port>` value is accepted and treated as broadcast with ID=0 for
+// backward compatibility.
+func parseEndpoint(str string) (*endpoint, error) {
```
Comment (adecaro) [inline platform/fabricx/core/membership/endpoint.go:37]: please, add unit-tests to make sure this function behave as expected especially in the corner cases.

**Review by adecaro — CHANGES_REQUESTED** [review]
Hi @tinystone007 , great catch and great effort.

Please, add unit-tests and we are good with this PR.

File: platform/fabricx/core/membership/endpoint.go (hunk 1 above)
Comment (tinystone007) [reply platform/fabricx/core/membership/endpoint.go:37]: Added unit tests in `platform/fabricx/core/membership/endpoint_test.go` (commit eff7d970) covering:

- `id=<n>,broadcast,host:port` and `id=<n>,deliver,host:port` happy paths
- multi-digit ID
- endpoint with embedded colons (IPv6-style `[::1]:7050`) round-tripping through the `.*` capture
- plain `host:port` falling through to the backward-compat fallback (`ID=0`, `Type=broadcast`)
- empty string → `ErrInvalidEndpointFormat` (asserted via `errors.Is`)
- ID overflowing `int` → wrapped `strconv` error

Let me know if there are other corner cases you'd like covered.

**Review by mbrandenburger — APPROVED** [review]
Thank you @tinystone007

**Review by adecaro — APPROVED** [review]
LGTM. Thanks a lot 🙏
